spec/src/main/asciidoc/architecture.asciidoc

```diff
@@ -1199,3 +1199,136 @@ The following metrics are produced for each channel declared by the application
+
+
+== Injecting a Publisher or PublishBuilder
+
+You can receive messages from a channel by injecting either a `Publisher` or `PublisherBuilder` and using the `@Channel` qualifier to specify the channel name:
+
+[source, java]
+----
+
+@ApplicationScoped
+public class BeanInjectedWithAPublisherOfPayloads {
+
+    private final Publisher<String> constructor;
+    @Inject
+    @Channel("hello")
+    private Publisher<String> field;
+
+    @Inject
+    public BeanInjectedWithAPublisherOfPayloads(@Channel("bonjour") Publisher<String> constructor) {
+        this.constructor = constructor;
+    }
+
+    public List<String> consume() {
+        return Flowable
+                .concat(
+                        Flowable.fromPublisher(constructor),
+                        Flowable.fromPublisher(field))
+                .toList()
+                .blockingGet();
+    }
+
+}
+
+----
+You can then inject `BeanInjectedWithAPublisherOfPayloads` to JAX-RS resources.
+
+The value _hello_ in the above example indicates the name of the channel.
+
+For a payload type `X`, the following types can be injected:
+
+* `Publisher<X>`
+* `PublisherBuilder<X>`
+* `Publisher<Message<X>>`
+* `Publisher<Message<X>>`
+
+
+== Publishing messages to a channel from imperative code
+
+Traditionally, the reactive world and imperative world are separated and operate in parallel. Reactive Messaging deals with streams of data in the reactive world, while the imperative world is pretty much point to point and synchronous communication. However, imperative programme sometimes needs to connect to reactive streams so that responses can be emitted to a destination service. 
+Bridging the two worlds is very valuable thing to do, so that one microservice can use technologies from both environments. For an instance, a JAX-RS resource might want to publish messages to a Reactive Messaging channel. This section is about enabling imperative code to publish messages to a Reactive Messaging channel, so that it can be consumed by a consumer.
+
+
+
+You can inject an `Emitter` and use it to send either payloads (`X`) or messages (`Message<X>`) to a channel as demonstrated below.
+
+[source, java]
+----
+@Inject @Channel("myChannel")
+private Emitter<String> emitter;
+
+public void publishMessage() {
+  emitter.send("a");
+  emitter.send("b");
+  emitter.complete();
+}
+----
+
+[source, java]
+----
+@Inject
+@Channel("foo")
+private Emitter<String> emitter;
+
+public void run() {
+    emitter.send(Message.of("a"));
+    emitter.send(Message.of("b"));
+    emitter.send(Message.of("c"));
+}
+
+----
+
+When injecting an `Emitter` (e.g. `@Inject Emitter`), you can configure how the back pressure is handled via `@OnOverflow` annotation, for the situation where emitting messages/payloads faster than the consumption of the messages.
```

Comment:
This whole section is now way better :+1:.

I'd just recommend this one addition to actually state that the `@Channel` qualifier must be used. It's implied by all the examples but unless I've missed it somewhere, we don't actually state it explicitly.

```suggestion
When injecting an `Emitter`, you must specify the target channel name using the `@Channel` qualifier.

When injecting an `Emitter`, you can configure how the back pressure is handled via `@OnOverflow` annotation, for the situation where emitting messages/payloads faster than the consumption of the messages.
```

Reply:
rephrased.